src/dump.c

```diff
@@ -1514,7 +1516,11 @@ static jl_value_t *jl_deserialize_value_module(jl_serializer_state *s)
     jl_sym_t *mname = (jl_sym_t*)jl_deserialize_value(s, NULL);
     int ref_only = read_uint8(s->s);
     if (ref_only) {
-        jl_value_t *m_ref = jl_get_global((jl_module_t*)jl_deserialize_value(s, NULL), mname);
+        jl_value_t *m_ref;
+        if (ref_only == 1)
+            m_ref = jl_get_global((jl_module_t*)jl_deserialize_value(s, NULL), mname);
+        else
+            m_ref = (jl_value_t*)jl_lookup_root_module((jl_value_t*)mname);
```

Comment:
What can I return from this function then?

Reply:
Er, you can't. All external state must be prepared by `read_verify_mod_list`, once `jl_deserialize_value` starts in earnest, this can only return existing references into that array.

Reply:
This doesn't change any global state --- it just looks up the module by name.

Reply:
This should just be using an index into `mod_array = jl_get_loaded_modules();` (as deserialized by `read_verify_mod_list`)

Reply:
Ok, I'll try that.